File: wlxx-services/wlxx-system/wlxx-system-provider/src/main/java/com/xzg/wlxx/system/controller/EmpController.java
package com.xzg.wlxx.system.controller;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.xzg.wlxx.common.base.ApiResult;
import com.xzg.wlxx.system.client.entity.dto.EmpDto;
import com.xzg.wlxx.system.client.entity.po.EmpPo;
import com.xzg.wlxx.system.service.EmpService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

/**
 * @author XiaoZG
 */
@RestController
@Tag(name = "员工")
@RequestMapping("emp")
@RequiredArgsConstructor
public class EmpController {

    private final EmpService service;

    @PostMapping("save")
    @Operation(summary = "新增员工")
    public ApiResult<Boolean> save(@RequestBody @Validated EmpDto dto) {
        return ApiResult.success(service.add(dto));
    }

    @GetMapping("searchByOrg")
    @Operation(summary = "根据组织查询员工")
    public ApiResult<IPage<EmpPo>> searchByOrg(EmpDto dto) {
        return ApiResult.success(service.searchByOrg(dto));
    }
}

File: wlxx-services/wlxx-system/wlxx-system-provider/src/main/kotlin/com/xzg/wlxx/system/controller/EmpController.kt
package com.xzg.wlxx.system.controller

import com.baomidou.mybatisplus.extension.plugins.pagination.Page
import com.xzg.wlxx.common.base.ApiResult
import com.xzg.wlxx.common.base.BaseController
import com.xzg.wlxx.system.entity.po.Emp
import com.xzg.wlxx.system.service.EmpService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.web.bind.annotation.*

@RestController
@Tag(name = "emp")
@RequestMapping("emp")
class EmpController(var service: EmpService) : BaseController() {

    @Operation(summary = "save")
    @PostMapping("save")
    fun save(@RequestBody emp: Emp): ApiResult<Boolean> {
        return success(service.insertOrUpdate(emp))
    }

    @Operation(summary = "page")
    @PostMapping("page/{page}/{size}")
    fun page(@RequestBody emp: Emp, @PathVariable page: Long, @PathVariable size: Long): ApiResult<Page<Emp>> {
        return success(service.page(Page(page, size)))
    }
}